extensions/wikia/AdEngine3/AdEngine3.setup.php

```diff
@@ -22,3 +21,16 @@
 // Hooks
 $wgHooks['OasisSkinAssetGroupsBlocking'][] = 'AdEngine3::onOasisSkinAssetGroupsBlocking';
 $wgHooks['WikiaSkinTopScripts'][] = 'AdEngine3::onWikiaSkinTopScripts';
+
+// Mirrors logic
+if ( !empty( $_SERVER['HTTP_FASTLY_FF'] ) && !empty($_SERVER[ 'X-Staging' ]) ) {
```

Comment:
What do you think about moving it to separate file `extensions/wikia/AdEngine3/AdHostMirrors.setup.php` and include new file in `includes/wikia/Extensions.php`. Then we don't need to duplicate this code in AdEngine2/3 extensions

Reply:
Perfection! ;-)